src/components/ColumnHeader/ColumnHeader.tsx

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+export enum SortDirection {
+  NONE = 'NONE',
+  ASC = 'ASC',
+  DESC = 'DESC',
+}
+
+export interface SortEvent {
+  name: string;
```

Comment:
I don't think there's a need of sending the `name` on this event. I believe this will be an unused prop. I'd rather just straight send the direction.

The reason behind this is that when the component is being used, you'd know which column you're controlling, because you're specifying a particular function.

Here's an example of usage of this component without sending the `name`:

```jsx
const Table = ({prop1, prop2}: TableProps) => {
  const [statusSortDirection, setStatusSortDirection] = useState<SortDirection>(SortDirection.NONE);
  const [createdSortDirection, setCreatedSortDirection] = useState<SortDirection>(SortDirection.NONE);

  return (
    <>
      {/* ... */}

      <ColumnHeader name="Status" onSort={setStatusSortDirection} />

      <ColumnHeader name="Created" onSort={setCreatedSortDirection} />

      {/* ... */}
    <>
  )
}

```
But sending the name in the sort event will only add boilerplate code, here's the same example but sending the name in the event:

```jsx
const Table = ({prop1, prop2}: TableProps) => {
  const [statusSortDirection, setStatusSortDirection] = useState<SortDirection>(SortDirection.NONE);
  const [createdSortDirection, setCreatedSortDirection] = useState<SortDirection>(SortDirection.NONE);

  return (
    <>
      {/* ... */}

      <ColumnHeader name="Status" onSort={({name, sortDirection}) => setStatusSortDirection(sortDirection))} />

      <ColumnHeader name="Created" onSort={({name, sortDirection}) => setCreatedSortDirection(sortDirection))} />

      {/* ... */}
    <>
  )
}
```